mods/persistence/game/machinery/biogenerator.dm

```diff
@@ -0,0 +1,39 @@
+/obj/machinery/biogenerator/Initialize()
+	. = ..()
+
+// remove and replace existing lists and their contents with persistence-friendly material
+	products = list(
+		"Food" = list(
+			/obj/item/chems/drinks/milk/smallcarton = 50,                   // 30u milk
+			/obj/item/chems/drinks/milk = 80,                               // 50u milk
+			/obj/item/chems/food/meat/syntiflesh = 100,                     // this can only be used for cutlets
+			/obj/item/chems/food/egg = 50,                                  // 3u egg yolk
+			/obj/item/storage/fancy/egg_box = 450),                         // 1 free egg
+		"Nutrients" = list(                                                 // this needs names really bad. you cant tell what's what until you get it.
+			/obj/item/chems/glass/bottle/eznutrient = 60,
+			/obj/item/chems/glass/bottle/left4zed = 100,
+			/obj/item/chems/glass/bottle/robustharvest = 120),
+		"Botanical Supplies" = list(
+			/obj/item/clothing/gloves/thick/botany = 250,                   // these are necessary for some plants so arent required via tailoring
+			/obj/item/storage/plants = 300,                                 // i would put this in tailoring but botany is hell without it
+			/obj/item/chems/spray/plantbgone = 250,                         // no chem reaction for this. needs to be. but keep this in here even if there is.
+			/obj/item/grenade/chem_grenade/antiweed = 1000),                // let them have fun, i say!
+		"Textiles" = list(
+			/obj/item/stack/material/bolt/mapped/cloth = 150,               // only source of clooth currently
+			/obj/item/stack/material/bolt/mapped/cloth/ten = 1500,
+//			/obj/item/stack/material/bolt/mapped/lace = 200,                // useless currently
+//			/obj/item/stack/material/bolt/mapped/lace/ten = 2000,
+//			/obj/item/stack/material/bolt/mapped/silk = 300,                // useless currently
+//			/obj/item/stack/material/bolt/mapped/silk/ten = 3000,
+//			/obj/item/stack/material/skin/mapped/synthleather =300,         // this actually can't be used outside of building literal walls or material armor, so I've commented it out for now so it's not a noob trap.
```

Comment:
you can also craft cloaks, banners, shoes, and boots with it, because it's a subtype of leather. it's also used to fabricate various things like belts, backpacks, briefcases, jackboots, workboots, synthleather coats, crime scene kits, xenoarchaeology belts, excavation pick sets, and fossil satchels.

Reply:
You're right. I double-checked it, and it does work on this codebase; I'll add it back to the list.